sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2065,6 +2065,16 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val OPTIMIZER_HINTS_ENABLED =
+    buildConf("spark.sql.optimizer.hints.enabled")
+      .internal()
+      .doc("Hints are additional directives that aids optimizer in better planning of a query. " +
+        "This configuration when set to `false`, disables the application of user " +
+        "specified hints.")
```

Comment:
nit: How about rephrasing it like this?
```
When false, the optimizer will ignore user-specified hints that are additional directives
for better planning of a query.
```

Reply:
@maropu Sounds good to me. Will change. Thank you.